docs/securing_installation.md

```diff
@@ -0,0 +1,108 @@
+
+1. Role-based access control, or RBAC
+2. Tiller's gRPC endpoint and its usage by Helm
+3. Tiller release information
+4. Helm charts
+
+### RBAC
+
+Recent versions of Kubernetes employ a [role-based access control (or RBAC)](https://en.wikipedia.org/wiki/Role-based_access_control) system as do modern operating systems to help mitigate the damage that can done if credentials are misused or bugs exist. Even in where an identity is hijacked, the identity has only so many permissions to a controlled space. This effectively adds a layer of security to limit the scope of any attack with that identity. 
+
+Helm and Tiller are designed to install, remove, and modify logical applications that can contain many service interacting together. As a result, often its usefulness involves cluster-wide operations, which in a multitenant cluster means that great care must be taken with access to a cluster-wide Tiller installation to prevent improper activity. 
+
+Specific users and teams -- developers, operators, system and network administrators -- will need their own portion of the cluster in which they can use Helm and Tiller without risking other portions of the cluster. This means using a Kubernetes cluster with RBAC enabled and Tiller configured to enforce them. For more information about using RBAC in Kubernetes, see [Using RBAC Authorization](rbac.md).
+
+#### Tiller and User Permissions
+Tiller in its current form (2.7.x) does not provide a way to map user credentials to specific permissions within Kubernetes. When Tiller is running inside of the cluster, it operates with the permissions of its service account. If no service account name is supplied to Tiller, it runs with the default service account. This means that all Tiller operations on that server are executed using the Tiller pod's credentials and permissions. 
+
+The properly limit what Tiller itself can do, the standard Kubernetes RBAC mechanisms must be attached to Tiller, including Roles and RoleBindings that place explicit limits on what things a Tiller instance can install, and where. 
+
+This situation may change in the future. While the community has several methods that might address this, at the moment performing actions using the rights of the client, instead of the rights of Tiller, is contingent upon the outcome of the Pod Identity Working Group, which has taken on the task of solving the problem in a general way. 
+
+
+### The Tiller gRPC Endpoint and TLS
+
+In the default installation the gRPC endpoint that Tiller offers is available inside the cluster (not external to the cluster) without authentication configuration applied. Without applying authentication, any process in the cluster can use the gRPC endpoint to perform operations inside the cluster. In a local or secured private cluster, this enables rapid usage and is normal. (From outside the cluster, Helm uses the normal authentication that Kubernetes supports to reach a gRPC endpoint.)
```

Comment:
The gRPC endpoint may be available outside of the cluster but on the same network depending on how networking is set up.  When users use a network config that bridges natively the the host network (for example, calico w/ BGP integration) the endpionts for gRPC will be accessible while the service endpoint (ClusterIP) will not.  This means it is still reachable.

Reply:
The line "from outside the cluster.." is a little misleading.  I'd say something like "When running outside the cluster the `helm` tool authenticates through the Kubernetes API server to reach helm and as such leverages existing Kuberenetes auth".

Reply:
@jbeda I think the key takeaway here is

> In the default installation

Are there any default cluster installations that support calico w/ BGP integration out of the box? I think we should be catering to the well-informed-yet-still-new user. If calico w/ BGP integration is an opt-in thing, these advanced users that are more well-informed or involved in their cluster setup will be more educated on whether the endpoint will be accessible outside of the cluster or not and can draw on decisions on their own. I'd rather keep this section simpler to give the 80% a better understanding.

Reply:
>  I'd say something like "When running outside the cluster the helm tool authenticates through the Kubernetes API server to reach helm and as such leverages existing Kuberenetes auth".

In the default case yes, but that's not always true. The client can bypass kube auth entirely by using `--host` or `$HELM_HOST`.

Reply:
@bacongobbler I believe that GKE, out of the box, has the cluster on the same network as VMs.  Calico on AWS also doesn't run encap but can configure routes.  Romana on AWS also does so.  Any kubenet networking will do so. This isn't at all uncommon.

Reply:
The networking issue is one that @lachie83 raised in conversation with me as well. I have a feeling that we'll have to seek more advice from the community here, while at the same time spelling out some of the areas in which attention must be paid, even if people in different scenarios do different things.